saleor/graphql/core/validators.py

```diff
@@ -42,3 +45,30 @@ def validate_price_precision(value: Optional["Decimal"], currency: str):
         raise ValidationError(
             f"Value cannot have more than {currency_fraction} decimal places."
         )
+
+
+def validate_variants_available_in_channel(
+    variants_id,
+    channel_id,
+    error_code,
+):
+    """Validate available variants in specific channel."""
+
+    available_variants = ProductVariantChannelListing.objects.filter(
+        variant__id__in=variants_id, channel_id=channel_id, price_amount__isnull=False
+    ).values_list("variant_id", flat=True)
+    not_available_variants = variants_id - set(available_variants)
+    if not_available_variants:
```

Comment:
`if not_available_variants := (variants_id - set(available_variants)):`

Reply:
This is copy from branch 3.0, I would like to avoid adding cosmetics changes, because this could cause conflicts later.